### PR TITLE
Parse and semantic-check min-elements and max-elements.

### PR DESCRIPTION
**Pull request by wenovus**
Currently, the statements don't undergo semantic checking, forcing
backends like ygot to have to handle it (which incidentally does it
wrong -- to be handled separately).

These fields are the most useful as uint64. This is because whether or
not they're explicitly set in the `list` or `leaf-list`, there is always
a valid uint64 value for them. As a result, uint64 is proposed as the
type for these fields.

By not using a pointer, however, the implementation of deviations
necessitates the new `deviationPresence` field to distiguish, for a
DeviationEntry, between no deviation applied vs. the default uint64
value. So, it has been added to the `Entry` type.

Lastly, deviation for deletes was actually unimplemented due to a test
bug that misinterpreted `nil` as "don't test it", rather than "test that
it's nil". This has been implemented and the test fixed.

### CODE REVIEW

File: pkg/yang/entry.go [hunk 1]
```diff
@@ -473,6 +485,30 @@ type DeviatedEntry struct {
...
+		return math.MaxUint64, nil
+	}
+	val, err := strconv.ParseUint(v.Name, 10, 64)
+	if err != nil {
+		return val, fmt.Errorf(`%s: invalid max-elements value %q (expect "unbounded" or a non-negative integer): %v`, Source(v), v.Name, err)
+	}
+	return val, nil
+}
+
+// semCheckMaxElements checks whether the min-element argument is valid, and returns the specified value.
```
[the remark this answers is not in the document]
Comment (wenovus) [reply pkg/yang/entry.go]: `go vet` is already there. Not sure which one is supposed to catch it.

**Comment by wenovus** [comment]
> I wonder what caused staticcheck removal?

Ah I will add it in a different PR.

File: pkg/yang/entry.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (robshakir) [reply pkg/yang/entry.go]: Interesting. I wonder whether this is because this is an internal API and `go vet` doesn't require a comment in that case.

**Comment by wenovus** [comment]
Hi @andaru, PTAL at https://github.com/openconfig/goyang/pull/125#pullrequestreview-433737444

**Pull request merged**